src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler\DoctrineOrmMappingsPass;
+use Doctrine\Common\Util\Inflector;
+use Symfony\Component\Config\Resource\DirectoryResource;
+use Symfony\Component\Config\Resource\FileResource;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * Load services stored in installed modules.
+ */
+class ModulesDoctrineCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $activeModules = $container->getParameter('kernel.active_modules');
+        $compilerPassList = $this->getCompilerPassList($activeModules);
+        /** @var CompilerPassInterface $compilerPass */
+        foreach ($compilerPassList as $compilerResourcePath => $compilerPass) {
+            $compilerPass->process($container);
+            if (is_dir($compilerResourcePath)) {
```

Comment:
if it's not dir, it's obvious a file..
```suggestion
            $container->addResource(
                is_dir($compilerResourcePath) ? 
                new DirectoryResource($compilerResourcePath)) :
                new FileResource($compilerResourcePath)
            );
```

Reply:
This is just a safe guard in case the file doesn't exist. It's not supposed to happen but I prefer to be careful as this could create problems during the install process.

Reply:
If the file does not exists the compiler pass will not return it. It's like parsing a directory with SymfonyFileSystem and adding a file_exists each time you try to get a file :thinking:

Reply:
That's right! Done